shell/platform/tizen/tizen_window_ecore_wl2.cc

```diff
@@ -0,0 +1,397 @@
+              delta_x += wheel_event->z;
+            }
+
+            self->view_->OnScroll(wheel_event->x, wheel_event->y, delta_x,
+                                  delta_y, kScrollOffsetMultiplier,
+                                  wheel_event->timestamp,
+                                  kFlutterPointerDeviceKindTouch, 0);
+            return ECORE_CALLBACK_DONE;
+          }
+        }
+        return ECORE_CALLBACK_PASS_ON;
+      },
+      this));
+
+  ecore_event_handlers_.push_back(ecore_event_handler_add(
+      ECORE_EVENT_KEY_DOWN,
+      [](void* data, int type, void* event) -> Eina_Bool {
+        auto* self = reinterpret_cast<TizenWindowEcoreWl2*>(data);
+        if (self->view_) {
+          auto* key_event = reinterpret_cast<Ecore_Event_Key*>(event);
+          if (key_event->window == self->GetWindowId()) {
+            self->view_->OnKey(key_event, true);
+            return ECORE_CALLBACK_DONE;
+          }
+        }
+        return ECORE_CALLBACK_PASS_ON;
+      },
+      this));
+
+  ecore_event_handlers_.push_back(ecore_event_handler_add(
+      ECORE_EVENT_KEY_UP,
+      [](void* data, int type, void* event) -> Eina_Bool {
+        auto* self = reinterpret_cast<TizenWindowEcoreWl2*>(data);
+        if (self->view_) {
+          auto* key_event = reinterpret_cast<Ecore_Event_Key*>(event);
+          if (key_event->window == self->GetWindowId()) {
+            self->view_->OnKey(key_event, false);
+            return ECORE_CALLBACK_DONE;
+          }
+        }
+        return ECORE_CALLBACK_PASS_ON;
+      },
+      this));
+}
+
+void TizenWindowEcoreWl2::UnregisterEventCallbacks() {
+  for (auto* handler : ecore_event_handlers_) {
+    ecore_event_handler_del(handler);
+  }
+  ecore_event_handlers_.clear();
+}
+
+void TizenWindowEcoreWl2::DestroyEcoreWl2() {
+  if (ecore_wl2_egl_window_) {
+    ecore_wl2_egl_window_destroy(ecore_wl2_egl_window_);
+    ecore_wl2_egl_window_ = nullptr;
+  }
+
+  if (ecore_wl2_window_) {
+    ecore_wl2_window_free(ecore_wl2_window_);
+    ecore_wl2_window_ = nullptr;
+  }
+
+  if (ecore_wl2_display_) {
+    ecore_wl2_display_disconnect(ecore_wl2_display_);
+    ecore_wl2_display_ = nullptr;
+  }
+  ecore_wl2_shutdown();
+}
+
+TizenWindow::Geometry TizenWindowEcoreWl2::GetWindowGeometry() {
+  Geometry result;
+  ecore_wl2_window_geometry_get(ecore_wl2_window_, &result.left, &result.top,
+                                &result.width, &result.height);
+  return result;
+}
+
+void TizenWindowEcoreWl2::SetWindowGeometry(Geometry geometry) {
+  ecore_wl2_window_geometry_set(ecore_wl2_window_, geometry.left, geometry.top,
+                                geometry.width, geometry.height);
+  // FIXME : I refer to https://github.com/flutter-tizen/engine/pull/201,
+  // but I don't know why it doesn't work properly without calling
+  // ecore_wl2_window_position_set().
+  ecore_wl2_window_position_set(ecore_wl2_window_, geometry.left, geometry.top);
+}
```

Comment:
`ecore_wl2_window_geometry_set` doesn't seems to change location of window. so I need calling `ecore_wl2_window_position_set`. 
maybe I misunderstood the behavior of `ecore_wl2_window_geometry_set`... how do you think about it?

Reply:
@bbrto21 
Would you try calling `ecore_wl2_window_commit(ecore_wl2_window_, EINA_TRUE)` after `ecore_wl2_window_geometry_set`?

I remember simply calling `ecore_wl2_window_geometry_set` didn't change the window size/position immediately, I called `ecore_wl2_window_position_set` because the API was already used in `TizenRenderEcoreWl2::SetupEcoreWl2` and it just worked.

I've checked implementation of `ecore_wl2_window_geometry_set`, it internally sets x, y, weight and height value in some config structure and marks some variable that geometry should be updated in the future. I believe `ecore_wl2_window_commit` is the function that applies the changes immediately (it sends update message to the compositor).

Reply:
@HakkyuKim it doesn't work... :(

Reply:
@bbrto21 
That is unfortunate..

Could you leave the comment as:
```
// FIXME: The changes set in `ecore_wl2_window_geometry_set` seems to apply only after calling `ecore_wl2_window_position_set`. Call a more appropriate API that flushes geometry settings to the compositor.
```

Reply:
@HakkyuKim I replaced it with the comment you left.

Reply:
The changes ... seems ... hmm.

How about: Currently ecore_wl2_window_position_set() must be called for the geometry settings to take effect. ...